### PR TITLE
Migrate etcd features from Spring Cloud and update etcd client

### PR DESCRIPTION
**Pull request by lewisheadden**
Takes https://github.com/spring-projects/spring-integration-extensions/pull/163 and updates the etcd client to be `com.ibm.etcd:etcd-java` which supports the etcd v3 API.

I think there are a few optimizations that could be made with Persistent Leases and other things in etcd v3 but I figured getting it working was a first step.

I also know that @dsyer's PR is fairly old so I may need to update deps but I'm not entirely sure what I need to update!

### CODE REVIEW

File: spring-integration-etcd/build.gradle [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+		}
+	}
+}
+
+sourceCompatibility = targetCompatibility = 1.7
+
+ext {
+	etcdVersion = '0.0.7'
+	slf4jVersion = '1.7.13'
+	springIntegrationVersion = '4.2.4.RELEASE'
```
Comment (artembilan) [inline spring-integration-etcd/build.gradle]: Please, upgrade to Spring Integration `5.0.8` (going to be `5.0.9` next week if that)

File: spring-integration-etcd/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#Sat Sep 01 18:17:28 EDT 2018
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.5-bin.zip
```
Comment (artembilan) [inline spring-integration-etcd/gradle/wrapper/gradle-wrapper.properties]: This is too old. Consider to upgrade to the latest one: https://docs.gradle.org/current/userguide/installation.html

**Comment by lewisheadden** [comment]
@artembilan thanks for your comments - they were very helpful in guiding me what things to update and what to drop. 

I think I managed to work through most of them. I upgraded Gradle Wrapper and fixed all the warnings there too (the `spring-integration-kafka` files were very useful to reference).

File: spring-integration-etcd/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+version: '3'
+services:
+  etcd:
+    image: quay.io/coreos/etcd:v3.3
+    ports:
+      - "2379:2379"
+      - "2380:2380"
+    command: /usr/local/bin/etcd --name spring-integration-etcd-test --advertise-client-urls http://0.0.0.0:2379 --listen-client-urls http://0.0.0.0:2379 --debug
```
[the remark this answers is not in the document]
Comment (lewisheadden) [reply spring-integration-etcd/docker-compose.yml]: I think in the previous PR, @dsyer had [comments on this](https://github.com/spring-projects/spring-integration-extensions/pull/163#discussion_r69881722) which I largely agree with. I think we need this config to allow people to run the integration tests locally easily. 

Further, the default `etcd` image isn't configured to bind to `0.0.0.0` but only to `localhost` ([by design](https://github.com/etcd-io/etcd/issues/8230#issuecomment-313799624)) so this took me a while to get worked out to get the tests running. I figured I would add the configuration here to make it as easy as `docker-compose up && ./gradlew test`.

[the remark this answers is not in the document]
Comment (artembilan) [reply spring-integration-etcd/docker-compose.yml]: Fair enough!

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration-extensions/commit/4f4d90376c497bcea0f66586ac8c48158539e088.

Although I see that I've squashed it so hard that only Dave's name left in the commit history.
I bring my apologies for such a negligence. I will ensure your next commit will be definitely with your name!

Thank you for contribution.
Looking forward for more!

**Comment by artembilan** [comment]
The artifact for testing is here: https://repo.spring.io/libs-snapshot-local/org/springframework/integration/spring-integration-etcd/1.0.0-BUILD-SNAPSHOT/

**Pull request closed without merging**